Tests/StoreKitUnitTests/PurchasesOrchestratorCommonTests.swift

```diff
@@ -495,4 +497,58 @@ class PurchasesOrchestratorTrackingTests: BasePurchasesOrchestratorTests {
         expect(mockDiagnosticsTracker.trackedPurchasesStartedParams.value).to(beEmpty())
         expect(mockDiagnosticsTracker.trackedPurchasesResultParams.value).to(beEmpty())
     }
+
+}
+
+#if !os(tvOS) && !os(watchOS) && compiler(>=5.10)
+
+// MARK: - Purchase Intent Received events
+
+@available(iOS 16.4, macOS 14.4, *)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@available(visionOS, unavailable)
+final class MockStoreKit2PurchaseIntent: StoreKit2PurchaseIntentType {
+
+    let product: StoreKit.Product
+    let offer: StoreKit.Product.SubscriptionOffer? = nil
```

Comment:
I think there's not an easy way to mock `StoreKit.Product.SubscriptionOffer` without introducing more unnecessary complexity, so for now I have this be `nil` in mocks, as we only need this for tracking the `apple_purchase_intent_received` event